pilot/pkg/security/authz/matcher/header.go

```diff
@@ -98,36 +96,21 @@ func HostMatcher(k, v string) *routepb.HeaderMatcher {
 		return &routepb.HeaderMatcher{
 			Name: k,
 			HeaderMatchSpecifier: &routepb.HeaderMatcher_StringMatch{
-				StringMatch: &matcher.StringMatcher{
-					IgnoreCase: true,
-					MatchPattern: &matcher.StringMatcher_Suffix{
-						Suffix: v[1:],
-					},
-				},
+				StringMatch: StringMatcherSuffix(strings.TrimPrefix(v, "*"), true),
 			},
 		}
 	} else if strings.HasSuffix(v, "*") {
 		return &routepb.HeaderMatcher{
 			Name: k,
 			HeaderMatchSpecifier: &routepb.HeaderMatcher_StringMatch{
-				StringMatch: &matcher.StringMatcher{
-					IgnoreCase: true,
-					MatchPattern: &matcher.StringMatcher_Prefix{
-						Prefix: v[:len(v)-1],
-					},
-				},
+				StringMatch: StringMatcherPrefix(strings.TrimSuffix(v, "*"), true),
```

Comment:
please keep the old behavior `v[:len(v)-1]`. This changes the behavior of `foo**` or similar.

Reply:
done, thanks.